### DISCUSSION TITLE
@apply with pseudo classes is not working as I expected

### DISCUSSION THREAD

**Discussion by adampatterson** [tailwindlabs/tailwindcss#3253] [Help]
I am using Laravel 8 with PostCSS, and Mix.

I am trying to set global styling in my app and I would like to specify the paragraph spacing.

I have done this by using:

```css
p {
    @apply pb-2
}
```

Next, I wanted to remove the padding on the `last-child` paragraph tag. 

So I enabled the variant.

```js
module.exports = {
    variants: {
        extend: {
            padding: ['last']
        }
    },
}
```

I then attempted to apply the pseudo class like this.

```css
p {
    @apply pb-2 last:pb-0;
}
```

Resulting in the following error:

```
(45:5) `@apply` cannot be used with `.last\:pb-0` because `.last\:pb-0` either cannot be found, or its actual definition includes a pseudo-selector like :hover, :active, etc. If you're sure that `.last\:pb-0` exists, make sure that any `@import` statements are being properly processed *before* Tailwind CSS sees your CSS, as `@apply` can only be used for classes in the same CSS tree.
```

So I tried a couple of other scenarios all resulting in the same error.

```css
p {
    @apply pb-2
    @apply last:pb-0;
}
```

```css
p {
    @apply last:pb-0;
}
```

For whatever reason, it looks like I am trying to apply `pb-0` as a pseudo-class to the class of `.last`.

Even if I do `<p class="last:pb-0">...</p>` there is still a `0.5rem` padding on the bottom.

The only thing that worked was to use the pseudo class in my CSS.

```css
p:last-child {
    @apply pb-0;
}
```

Am I loading the variant properly?

Am I using `@apply` wrong?

Thanks!

**Comment by adampatterson** [comment]
I am still looking for a "tailwind" solution, I came across `applyComplexClasses` and updated Tailwind to 1.9, but it does not seem to work with last.

```js
module.exports = {
  experimental: {
    applyComplexClasses: true,
  },
}
```

```
(48:21) The `last:pb-0` class does not exist. If you're sure that `last:pb-0` exists, make sure that any `@import` statements are being properly processed before Tailwind CSS sees your CSS, as `@apply` can only be used for classes in the same CSS tree.
```

The error is different and it now looks like `last:pb-0` is correct

**Comment by adamwathan** [comment]
Make sure you are on Tailwind 2.0, otherwise the “extend” under variants won’t work. You won’t need that experimental flag in 2.0 either 👍🏻

**Comment by adampatterson** [comment]
OK, I am using 1.9.6 I think. That might explain it all then :)

I will give it a try in the morning then!

**Comment by Boanerges-Vaca** [comment]
`Error: Expected a pseudo-class or pseudo-element.
    at D:\repo .net\AppTailwind\src\styles.scss:11:9`

Tengo este Error al Personalizar con scss.

 ` @layer components {

    .btn {
        @apply py-2 px-4 rounded font-bold text-sm;
    }
    .btn-prim {
        @apply text-white focus: outline-none bg-blue-500 hover: bg-blue-600 hover: shadow-lg;
    }
} `

con una sola linea personalizacion no hay error

` @layer components {

    .btn {
        @apply py-2 px-4 rounded font-bold text-sm;
    }
 `
